service/share/full_availability_test.go

```diff
@@ -35,10 +37,40 @@ func TestShareAvailableOverMocknet_Full(t *testing.T) {
 	defer cancel()
 
 	net := NewDAGNet(ctx, t)
-	_, root := net.RandFullService(16)
-	serv := net.CleanService()
+	_, root := net.RandFullService(32)
+	serv := net.CleanFullService()
 	net.ConnectAll()
 
 	err := serv.SharesAvailable(ctx, root)
 	assert.NoError(t, err)
 }
+
+func TestShareAvailable_FullOverLights(t *testing.T) {
+	const (
+		origSquareSize = 8
+		lightNodes     = 192
+	)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	net := NewDAGNet(ctx, t)
+	_, root := net.RandFullService(origSquareSize) // make a source node, a.k.a bridge
+	full := net.CleanFullService()                 // make a full which reconstructs data
+
+	lights := make([]Service, lightNodes)
+	for i := 0; i < len(lights); i++ {
+		lights[i] = net.CleanLightService()
+	}
+	net.ConnectAll()
+	for i := 0; i < len(lights); i++ {
+		err := lights[i].SharesAvailable(ctx, root)
+		require.NoError(t, err)
+	}
+
+	// ensure there is no connection between source and full nodes
+	// so that full reconstructs from the light nodes only
+	net.Disconnect(0, 1)
```

Comment:
Q: This means that the fullnode is connected to _all_ light nodes though, right?

Reply:
There is a connect all above

Reply:
I'm wondering if we should not also test the case where that full node is not directly connected to all light nodes directly. Also, we should have a test where there are a few full nodes that are each connected to a (potenially overlapping) subset of light nodes.
But this test is already super cool!

Reply:
We can do more such tests, sure.  

About the case with full not connected to all light clients, how is it supposed to find others?

Reply:
Now there is also `TestShareAvailable_MultipleFullOverLights` which is what I meant. 

> About the case with full not connected to all light clients, how is it supposed to find others?

That is a good question. I think we should capture this in an issue if it does not exist already. IIRC, we are now assuming that the light nodes find the full nodes and not vice versa, right?

Reply:
Yes, so you can resolve this comment now until we implement discovery.(https://github.com/celestiaorg/celestia-node/issues/599)
> IIRC, we are now assuming that the light nodes find the full nodes and not vice versa, right?